tests/eos-test-utils.c

```diff
@@ -164,6 +170,7 @@ eos_test_subserver_dispose_impl (EosTestSubserver *subserver)
 {
   g_clear_pointer (&subserver->devices, g_ptr_array_unref);
   g_clear_pointer (&subserver->ref_to_commit, g_hash_table_unref);
+  g_clear_pointer (&subserver->detached_metadata, g_hash_table_unref);
```

Comment:
I think you can remove all the detached metadata harness, since we're not going to use it for this feature after all